ui/src/components/QuickSetup.tsx

```diff
@@ -0,0 +1,598 @@
+import React, { useEffect, useState } from "react"
+import { Form, Button, Grid, Loader, Table } from "semantic-ui-react"
+import { useHistory } from "react-router-dom"
+
+import DamlLedger, { useLedger, useParty } from "@daml/react"
+
+import Ledger from "@daml/ledger"
+
+import { PartyDetails, WellKnownPartiesProvider } from "@daml/hub-react"
+import { retrieveParties } from "../Parties"
+
+import { BrokerInvitation } from "@daml.js/da-marketplace/lib/Marketplace/Broker"
+import { CustodianInvitation } from "@daml.js/da-marketplace/lib/Marketplace/Custodian"
+import { ExchangeInvitation } from "@daml.js/da-marketplace/lib/Marketplace/Exchange"
+import { InvestorInvitation } from "@daml.js/da-marketplace/lib/Marketplace/Investor"
+import { IssuerInvitation } from "@daml.js/da-marketplace/lib/Marketplace/Issuer"
+
+import { MarketRole } from "@daml.js/da-marketplace/lib/Marketplace/Utils"
+
+import { UserSession } from "@daml.js/da-marketplace/lib/Marketplace/Onboarding"
+import { User } from "@daml.js/da-marketplace/lib/Marketplace/Onboarding"
+
+import { RegistryLookupProvider, useRegistryLookup } from "./common/RegistryLookup"
+import { halfSecondPromise } from "./common/utils"
+import { roleLabel } from "./common/utils"
+
+import QueryStreamProvider, {
+    useContractQuery,
+    usePartyLoading,
+    useLoading,
+} from "../websocket/queryStream"
+
+import { useDablParties, useOperator } from "../components/common/common"
+
+import Credentials, { retrieveCredentials } from "../Credentials"
+
+import { httpBaseUrl, deploymentMode, DeploymentMode } from "../config"
+
+import deployTrigger, { TRIGGER_HASH, MarketplaceTrigger } from "../../src/automation"
+
+interface IPartyLoginData extends PartyDetails {
+    role: MarketRole
+    name: string
+    location: string
+    title?: string
+    issuerId?: string
+    deployMatchingEngine?: boolean
+}
+
+const MARKETROLES = ["CustodianRole", "IssuerRole", "ExchangeRole", "InvestorRole", "BrokerRole"]
```

Comment:
I think the `MarketRole` enum values would work here directly as well (they type just fine to `string`).

Reply:
oh thanks!